### ISSUE TITLE
train.py stops unexpectedly

### ISSUE THREAD

**Issue by GiacomoPorpiglia** [bmdanielsson/nnue-trainer#2]
Hello, can you use .binpack format instead of .bin ? 
I haven't tried the program with .bin data, but with .binpack it just stops and doesn't enter the loop in train.py at line 165. Therefore, here seems to be a problem with train_data_loader, but I don't know if the file extension is the reason.

Edit: I have also tried using .bin data, but the problem persists. I don't know what the problem is, here is the log I get before the program stops.

Device: cuda:0
Training set: train_data.bin
Validation set: val_data.bin
Batch size: 16384
Using factorizer: False
Lambda: 1.0
Validation check interval: 2000
Data written to: output/7

Ranger optimizer loaded.
Gradient Centralization usage = True
GC applied to both conv and fc layers

**Comment by bmdanielsson** [comment]
Hi,

The binpack format is not supported, but bin format should work. How large is your .bin file? Would it be possible for you to share it?

Regards
/Martin

**Comment by GiacomoPorpiglia** [comment]
Hello, thanks for your response. By serching the problem online, it seems that the file was too big for my RAM or GPU memory.

**Comment by bmdanielsson** [comment]
The file size shouldn't be an issue since the dataloader only works with 100000 entries (~4MB) at a time. However the batch size of 16384 could be an issue if you have a GPU with little memory. Try setting a smaller batch size and see if that helps, for instance 
'--batch-size 1024' .
